contracts/src/assertionStakingPool/EdgeStakingPool.sol

```diff
@@ -0,0 +1,51 @@
+// Copyright 2021-2022, Offchain Labs, Inc.
+// For license information, see https://github.com/OffchainLabs/nitro-contracts/blob/main/LICENSE
+// SPDX-License-Identifier: BUSL-1.1
+
+pragma solidity ^0.8.0;
+
+import "./AbsBoldStakingPool.sol";
+import "../challengeV2/EdgeChallengeManager.sol";
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/// @notice Staking pool contract for creating layer zero edges.
+/// Allows users to deposit stake, create an edge once required stake amount is reached,
+/// and reclaim their stake when and if the edge is confirmed.
+/// @dev Unlike the assertion staking pool, there is no need for a function to claim the stake back into the pool.
+contract EdgeStakingPool is AbsBoldStakingPool {
+    using SafeERC20 for IERC20;
+
+    EdgeChallengeManager public immutable challengeManager;
+    bytes32 public immutable createEdgeArgsHash;
+    uint256 public immutable requiredStake;
+
+    /// @notice The provided arguments to not match createEdgeArgsHash
+    error IncorrectCreateEdgeArgs();
+
+    /// @param _challengeManager EdgeChallengeManager contract
+    /// @param createEdgeArgs Arguments to be passed into EdgeChallengeManager.createLayerZeroEdge
+    constructor(
+        address _challengeManager,
+        CreateEdgeArgs memory createEdgeArgs
+    ) AbsBoldStakingPool(EdgeChallengeManager(_challengeManager).stakeToken()) {
+        challengeManager = EdgeChallengeManager(_challengeManager);
+        createEdgeArgsHash = keccak256(abi.encode(createEdgeArgs));
```

Comment:
Why we only save the hash here, instead similar to `AssertionStakingPool` we store the whole struct?
Consider to align both, I think we should store only the hash in both case

Reply:
hashing both sounds good to me

Reply:
i avoided storing the `CreateEdgeArgs` because of the large ish proofs